examples/features/standard/broker-plugin/src/main/java/org/apache/activemq/artemis/jms/example/BrokerPlugin.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.jms.example;
+
+import org.apache.activemq.artemis.api.core.ActiveMQException;
+import org.apache.activemq.artemis.api.core.ICoreMessage;
+import org.apache.activemq.artemis.api.core.Message;
+import org.apache.activemq.artemis.core.server.ServerSession;
+import org.apache.activemq.artemis.core.server.plugin.ActiveMQServerPlugin;
+import org.apache.activemq.artemis.core.transaction.Transaction;
+import org.apache.activemq.artemis.protocol.amqp.broker.AMQPMessage;
+
+public class BrokerPlugin implements ActiveMQServerPlugin {
+
+   int count = 0;
+
+   @Override
+   public void beforeSend(ServerSession session,
+                          Transaction tx,
+                          Message message, boolean direct,
+                          boolean noAutoCreateQueue) throws ActiveMQException {
+      if (message instanceof AMQPMessage) {
+         // AMQP Messages are by definition immutable.
```

Comment:
I don't think you need to treat these differently based on the type of message. I think you can just say 
`message.putStringProperty()
message.reencode()`
If i remember correctly, ICoreMessage just inherits the default noop implementation of `reencode`

Reply:
@harrison-tarr  that's right... and do you believe I wrote that? :)